core/src/components/checkbox/test/states/checkbox.e2e.ts

```diff
@@ -0,0 +1,54 @@
+import { expect } from '@playwright/test';
+import { configs, test } from '@utils/test/playwright';
+
+configs({ directions: ['ltr'] }).forEach(({ title, screenshot, config }) => {
+  test.describe(title('checkbox: states'), () => {
+    test('should render disabled checkbox correctly', async ({ page }) => {
+      await page.setContent(
+        `
+          <ion-checkbox disabled>Label</ion-checkbox>
+      `,
+        config
+      );
+
+      const checkbox = page.locator('ion-checkbox');
+      expect(await checkbox.screenshot()).toMatchSnapshot(screenshot(`checkbox-disabled`));
+    });
+
+    test('should render disabled checked checkbox correctly', async ({ page }) => {
+      await page.setContent(
+        `
+          <ion-checkbox checked disabled>Label</ion-checkbox>
+      `,
+        config
+      );
```

Comment:
Do we need to capture both checked and disabled? If we're applying the opacity to the host, I'm not sure we gain anything significant by testing the checked variant too.

Reply:
I was thinking we just needed a screenshot of the checked state when disabled to verify the color for any future updates.

Reply:
I think it's more-so making sure the checkbox still looks right when enabled. Avoiding any regressions down the road.